rpc/transport/transport.go

```diff
@@ -22,28 +22,31 @@ import (
 // with RecvMessage.
 type Transport interface {
 	// NewMessage allocates a new message to be sent over the transport.
-	// The caller must call the release function when it no longer needs
-	// to reference the message. Calling the release function more than once
-	// has no effect. Before releasing the message, send may be called at most
-	// once to send the mssage.
+	// The caller must call OutgoingMessage.Release() when it no longer
+	// needs to reference the message. Calling Release() more than once
+	// has no effect.  Before releasing the message, Send() MAY be called
+	// at most once to send the mssage.
 	//
-	// Messages returned by NewMessage must have a nil CapTable.
-	// When the returned ReleaseFunc is called, any clients in the message's
-	// CapTable will be released.
+	// OutgoingMessage instances returned by NewMessage must have a nil
+	// CapTable.  When OutgoingMessage.Release() is called, the underlying
+	// *capnp.Message will be released. This will also release any clients
+	// in the message's CapTable and release its Arena.
 	//
 	// The Arena in the returned message should be fast at allocating new
```

Comment:
Not really related to this PR (no need to address here) but: Is there actually a reason for this `should`? Afaik there's nothing in the RPC subsystem that would make SingleSegment a problem.

Reply:
I always assumed it was because the `PlaceArgs` call might allocate lots of (potentially large) structs?  In other words,  it would be a case of "we're handing off control the user, so make sure we can handle whatever they do". 🤷‍♂️